addons/point_of_sale/models/pos_config.py

```diff
@@ -370,8 +380,16 @@ def create(self, values):
 
     @api.multi
     def write(self, vals):
-        result = super(PosConfig, self).write(vals)
+        if self.env['pos.session'].search_count([('config_id', '=', self.id), ('state', '!=', 'closed')]):
+            forbidden_fields = []
+            for check_field in self._get_fields_to_check():
+                if check_field.get('field') in vals:
+                    del vals[check_field.get('field')]
```

Comment:
No need to bother deleting in vals. If you have at least one forbidden field, an error will be raised, the `super` won't be called and the transaction will be rollbacked.